cmd/admin-handlers.go

```diff
@@ -450,26 +481,73 @@ func (a adminAPIHandlers) GetConfigHandler(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	// Take a read lock on minio/config.json. NB minio is a
-	// reserved bucket name and wouldn't conflict with normal
-	// object operations.
-	configLock := globalNSMutex.NewNSLock(minioReservedBucket, minioConfigFile)
-	if configLock.GetRLock(globalObjectTimeout) != nil {
-		writeErrorResponseJSON(w, ErrOperationTimedOut, r.URL)
+	configBytes, err := a.getConfigDotJSON()
+	if err != nil {
+		logger.LogIf(context.Background(), err)
+		writeErrorResponseJSON(w, toAdminAPIErrCode(err), r.URL)
+	}
+
+	writeSuccessResponseJSON(w, configBytes)
+}
+
+// Disable tidwall json array notation in JSON key path so
+// users can set json with a key as a number.
+//    In tidwall json, notify.webhook.0 = val means { "notify" : { "webhook" : [val] }}
+//    In Minio, notify.webhook.0 = val means { "notify" : { "webhook" : {"0" : val}}}
+func normalizeJSONKey(input string) (key string) {
+	subKeys := strings.Split(input, ".")
+	for i, k := range subKeys {
+		if i > 0 {
+			key += "."
+		}
+		if _, err := strconv.Atoi(k); err == nil {
+			key += ":" + k
+		} else {
+			key += k
+		}
+	}
+	return
+}
+
+// GetConfigHandler - GET /minio/admin/v1/config-keys
+// Get some keys in config.json of this minio setup.
+func (a adminAPIHandlers) GetConfigKeysHandler(w http.ResponseWriter, r *http.Request) {
+	// Validate request signature.
+	adminAPIErr := checkAdminRequestAuthType(r, "")
+	if adminAPIErr != ErrNone {
+		writeErrorResponseJSON(w, adminAPIErr, r.URL)
 		return
 	}
-	defer configLock.RUnlock()
 
-	// Get config.json - in distributed mode, the configuration
-	// occurring on a quorum of the servers is returned.
-	configBytes, err := getPeerConfig(globalAdminPeers)
+	var keys []string
+	queries := r.URL.Query()
+
+	for k := range queries {
+		keys = append(keys, k)
+	}
+
+	configBytes, err := a.getConfigDotJSON()
 	if err != nil {
 		logger.LogIf(context.Background(), err)
 		writeErrorResponseJSON(w, toAdminAPIErrCode(err), r.URL)
 		return
 	}
 
-	writeSuccessResponseJSON(w, configBytes)
+	configStr := string(configBytes)
+	jsonResult := `{}`
+	for _, key := range keys {
+		// sjson.Set does not return an error if key is empty
+		// we should check by ourselves here
+		if key == "" {
+			continue
+		}
+		val := gjson.Get(configStr, key)
+		if j, err := sjson.Set(jsonResult, normalizeJSONKey(key), val.Value()); err == nil {
+			jsonResult = j
```

Comment:
Seems like we ignore the erorr here, which is fine. But I am not sure how we can pass multiple keys and even if we can, it seems like we override `jsonResult` with the latest result. Is this intended?

Reply:
What is the intended use case here? Can we send multiple keys here, if yes, how do we return the resuls here.

Reply:
> Seems like we ignore the erorr here, which is fine.

Yeah, Ersan & I decided to merge this first, then Ersan will send a new PR with an extensive error validation that he already did.

> it seems like we override jsonResult with the latest result. Is this intended?

If you see jsonResult starts with '{}' and in the loop, it will get populated with the value of each passed key.

Reply:
`jsonResult = j`, means you are overwriting, the previous value

Reply:
So, 
`jsonResult = j` => `jsonResult += j`

Reply:
> So, the fix is:
> jsonResult = j => jsonResult += j

No,

```
if j, err := sjson.Set(jsonResult, normalizeJSONKey(key), val.Value()); err == nil {
			jsonResult = j
}
```

sjson.Set() will already add the new key/value to jsonResult, so `jsonResult = j` means jsonResult will receive the updated json.

Reply:
ok